usaspending_api/submissions/management/commands/load_submission.py

```diff
@@ -309,22 +301,67 @@ def handle(self, *args, **options):
-                continue
+            # Try and find the parent award
+            award = None
+            try:
+                award = find_or_create_summary_award(piid=row.get('piid'), fain=row.get('fain'), uri=row.get('uri'), parent_award_id=row.get('parent_award_id'))
+            except:
+                self.logger.info("Could not find parent award with piid " + row.get('piid', '') + " fain " + row.get('fain', '') + " uri " + row.get('uri', '') + " parent " + row.get('parent_award_id', ''))
 
             procurement_value_map = {
                 "award": award,
-                'submission': submission_attributes
+                'submission': submission_attributes,
+                "action_date": datetime.strptime(row['action_date'], '%Y%m%d')
             }
 
-            procurement = load_data_into_model(Procurement(), row, value_map=procurement_value_map, save=True)
+            load_data_into_model(Procurement(), row, value_map=procurement_value_map, save=True)
+
+
+def get_treasury_appropriation_account_tas_lookup(tas_lookup_id, db_cursor):
+    # Checks the broker DB tas_lookup table for the tas_id and returns the matching TAS object in the datastore
+    db_cursor.execute('SELECT * FROM tas_lookup WHERE tas_id = %s', [tas_lookup_id])
+    tas_data = dictfetchall(db_cursor)
+
+    # These or "" convert from none to a blank string, which is how the TAS table stores nulls
+    q_kwargs = {
+        "allocation_transfer_agency_id": tas_data[0]["allocation_transfer_agency"] or "",
+        "agency_id": tas_data[0]["agency_identifier"] or "",
+        "beginning_period_of_availability": tas_data[0]["beginning_period_of_availability"] or "",
+        "ending_period_of_availability": tas_data[0]["ending_period_of_availability"] or "",
+        "availability_type_code": tas_data[0]["availability_type_code"] or "",
+        "main_account_code": tas_data[0]["main_account_code"] or "",
+        "sub_account_code": tas_data[0]["sub_account_code"] or ""
+    }
+
+    return TreasuryAppropriationAccount.objects.filter(Q(**q_kwargs)).first()
+
+
+def find_or_create_summary_award(piid=None, fain=None, uri=None, parent_award_id=None):
+    # If an award's ID is a piid, it's parent is a PIID; likewise for URI
+    # Awards with FAINs don't have parents
+    summary_award = None
+    q_kwargs = {}
+    for i in [(piid, "piid"), (uri, "uri"), (fain, "fain")]:
+        if i[0]:
+            q_kwargs[i[1]] = i[0]
+            if parent_award_id:
+                q_kwargs["parent_award__" + i[1]] = parent_award_id
+            else:
+                q_kwargs["parent_award"] = None
+
+            # Now search for it
+            summary_award = Award.objects.all().filter(Q(**q_kwargs)).first()
+            if summary_award:
+                return summary_award
+            else:
+                # If we can't find it, we need to make it. Recursively get the parent
+                parent_award = None
```

Comment:
See earlier questions about parent award and whether or not we should be using for this.